examples/sveltekit-openai/src/routes/chat/[id]/+page.svelte

```diff
@@ -0,0 +1,140 @@
+<script lang="ts">
+  import { page } from '$app/state';
+  import ArrowUp from '$lib/components/icons/arrow-up.svelte';
+  import Button from '$lib/components/ui/button/button.svelte';
+  import { Textarea } from '$lib/components/ui/textarea/index.js';
+  import { Chat } from '@ai-sdk/svelte';
+  import { defaultChatStore } from 'ai';
+
+  const chat = new Chat(() => ({
```

Comment:
@elliott-with-the-longest-name-on-github the chat class tests pass but i'm seeing funky behavior w/ re-rendering message part changes when running the example!

if you have time to look or if you have time to sync again tomorrow lmk!

Reply:
seems like under the hood, sveltekit's optimizations for only re-rendering when state changes doesn't account for the deeply nested message parts changes

Reply:
(for context: chat store's getMessages just returns chat.messages!)